base/ltfiles.dtx

```diff
@@ -1251,8 +1252,23 @@
   \expandafter\def\expandafter\reserved@a
     \expandafter{\reserved@a{#2}{#3}}%
 \reserved@a}
+%    \end{macrocode}
+%    Pipes are not really files, but at the document level they are supported.
+%    To quickly trim of any leading spaces, we use a blank test and \cs{use:n}
+%    rather than \cs{tl_trim_spaces:n} for speed as we don't care about the end
+%    of the input.
+%    \begin{macrocode}
 \ExplSyntaxOn
-\cs_new_eq:NN \IfFileExists@@ \file_full_name:n
+\cs_new:Npn \IfFileExists@@ #1
+  {
+    \tl_if_blank:nF {#1}
+      {
+        \tl_if_head_eq_charcode:eNTF { \use:n #1 } |
+          {#1}
+          { \file_full_name:n {#1} }
+      }
+  }
+\cs_generate_variant:Nn \tl_if_head_eq_charcode:nNTF { e }
```

Comment:
`oNTF` is enough (and slightly quicker?).

The `#1` of both `\IfFileExists@` and `\IfFileExists@@` is the content of `\@curr@file`, hence already expanded.
https://github.com/latex3/latex2e/blob/22c47f043fcf5d446cd75797846f9e3b008d3b13/base/ltfiles.dtx#L1162-L1164